ports/rp2/pendsv.c

```diff
@@ -97,13 +135,19 @@ void pendsv_schedule_dispatch(size_t slot, pendsv_dispatch_t f) {
 // PendSV interrupt handler to perform background processing.
 void PendSV_Handler(void) {
 
+    #if MICROPY_PY_THREAD
     if (!mp_thread_recursive_mutex_lock(&pendsv_mutex, 0)) {
         // Failure here means core 1 holds pendsv_mutex. ISR will
         // run again after core 1 calls pendsv_resume().
         return;
     }
     // Core 0 should not already have locked pendsv_mutex
     assert(pendsv_mutex.mutex.enter_count == 1);
+    #else
+    if (pendsv_suspend_count() > 0) {
+        return; // Will run again after pendsv_resume()
```

Comment:
The original implementation simply had `assert(pendsv_lock == 0)` here, which is much stricter.  Did you find cases where this handler can be called with `pendsv_lock > 0` ?

Reply:
Probably not on single threaded, assuming the PendSV runs without latency. Will change back to the assert.